File: selling/report/employee_wise_sales_order_report/employee_wise_sales_order_report.py
# Copyright (c) 2013, Web Notes Technologies Pvt. Ltd. and Contributors
# License: GNU General Public License v3. See license.txt

from __future__ import unicode_literals
import webnotes
#from webnotes import _, msgprint
from webnotes.utils import flt
#sql = webnotes.conn.sql


from webnotes import _, msgprint

def execute(filters=None):
        if not filters: filters = {}

        columns = get_columns()
        data = get_details(filters)
        return columns,data

def get_columns():
        columns=["Region::100" ,"Employee::150","Sales Order No.::150","Order Date::100","OEM/Direct:Select:100", "Product::120" , "Amount::120","Sales Commission::120","Invoice Date::120" ]
        return columns

def get_conditions(filters):
        conditions = ""
        if filters.get("territory"):
            conditions += "and a.territory='%s'"%filters.get("territory")
        if filters.get("owner"):
            conditions += "and a.owner='%s'"%filters.get("owner")
        if filters.get("transaction_date"):
            conditions += "and a.transaction_date='%s'"%filters.get("transaction_date")
        if filters.get("creation"):
            conditions += "and soi.creation='%s'"%filters.get("creation")
        if filters.get("customer_type"):
            conditions += "and a.customer_type='%s'"%filters.get("customer_type")
        if filters.get("item_code"):
            conditions += "and x.item_code='%s'"%filters.get("item_code")
        if filters.get("net_total_export"):
            conditions += "and a.net_total_export='%s'"%filters.get("net_total_export")
        if filters.get("total_commission"):
            conditions += "and a.total_commission='%s'"%filters.get("total_commission")
        return conditions


def get_details(filters):
        conditions = get_conditions(filters)
        return  webnotes.conn.sql("""select a.territory,a.owner,a.name,a.transaction_date,a.customer_type,x.item_code, a.net_total_export,a.total_commission,date(soi.creation) from
`tabSales Order` a
left join (select item_code,parent from `tabAccessories Details` b union select item_code,parent from `tabSales Order Item` c ) x on x.parent=a.name
left join (select creation,sales_order from `tabSales Invoice Item` si where creation in(select min(creation) from `tabSales Invoice Item` where sales_order=si.sales_order)) soi on soi.sales_order=a.name 
order by owner,territory %s"""%conditions, as_list=1)

File: selling/report/employee_wise_sales_order_report/employee_wise_sales_order_report.js
// Copyright (c) 2013, Web Notes Technologies Pvt. Ltd.
// License: GNU General Public License v3. See license.txt

wn.query_reports["Employee Wise Sales Order Report"] = {
        "filters": [
                {
                        "fieldname":"territory",
                        "label": "Region",
                        "fieldtype": "Link",
                        "options": "Territory",
                        //"default": wn.defaults.get_user_default("raw_item")
                },
                {
                        "fieldname":"owner",
                        "label": "Employee",
                        "fieldtype": "Link",
                        "options": "Profile",
                        //"default": wn.defaults.get_user_default("rawitem_warehouse")
                },
		{
                        "fieldname":"transaction date",
                        "label": " From Date",
                        "fieldtype": "Date",
                        //"options": "Item",
                        //"default": wn.defaults.get_user_default("product_code")
                },
                {
                        "fieldname":"creatoin",
                        "label": "To Date",
                        "fieldtype": "Date",
                        //"options": "Batch",
                        //"default": wn.defaults.get_user_default("select_batchno")
                },
                {
                        "fieldname":"customer_type",
                        "label": "OEM",
                        "fieldtype": "Select",
                        "options": ["","OEM","Direct"],
                        //"default": wn.defaults.get_user_default("select_batchno")
                },
                {
                        "fieldname":"item_code",
                        "label": "Product",
                        "fieldtype": "Link",
                        "options": "Item",
                        //"default": wn.defaults.get_user_default("select_batchno")
                },
		{
                        "fieldname":"net_total_export",
                        "label": "Amount",
                        "fieldtype": "Currency",
                        //"options": "Batch",
                        //"default": wn.defaults.get_user_default("select_batchno")
                },
                {
                        "fieldname":"total_commission",
                        "label": "Sales_Commission",
                        "fieldtype": "Currency",
                        //"options": "Batch",
                        //"default": wn.defaults.get_user_default("select_batchno")
                },

        ]
}
